core/server/common/src/main/java/alluxio/master/journal/CheckpointOutputStream.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.master.journal;
+
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+
+/**
+ * Output stream for writing checkpoints.
+ *
+ * The stream begins with a version number.
```

Comment:
I'm not sure what this version number is supposed to mean and how it is used. Is it related to the `JOURNAL_ENTRY_CHECKPOINT_VERSION`? Can different checkpoint names/types have different versions?